File: nils-tdt-4100/foreksempel/src/main/java/uke2/tell_ned.py



def tell_ned(tall:int) -> int:
    return tall - 1


tall = 6
tall = tell_ned(tall)
print(tall)
tall = tell_ned(tall)
print(tall)
tall = tell_ned(tall)
print(tall)
tall = tell_ned(tall)
print(tall)

# Her viser jeg hvordan typesjekking kan innføres i Python også:
# Det er derimot lov til å kjøre filen, og den vil kjøre helt til en kommer til den linjen.
tall = tell_ned("ost") # https://www.emmanuelgautier.com/blog/enable-vscode-python-type-checking

# Disse funksjonene har ingen kontekst.
# Sett at du har noe som skal løses i en helt bestemt omgivelse.
# Hva hvis jeg vil telle ned flere ulike ting på en gang?
# Hva er beste måten å lage flere slike tellere?

File: nils-tdt-4100/foreksempel/src/main/java/uke2/TellNed.java
package uke2;

public class TellNed {

    // Hvilke indre 'data' er det klassen TellNed må holde rede på?
    // Disse 'attributtene' *bør* defineres før man går i gang med å definere metoder.
    int tall;

    // Konstruktør. Denne kalles når man lager et nytt objekt av denne klassen.
    // Man kan ha flere av dem så lenge de har ulike parametre.
    public TellNed(int tall) {
        this.tall = tall;
    }

    // Metode for å telle ned i et TellNed-objekt.
    public void tellned() {
        tall -= 1;
    }
    
    // Metode 
    public int getTall() {
        return tall;
    }

    public static void main(String[] args) {
        TellNed tn1 = new TellNed(10);
        tn1.tellned();
        System.out.println("tn1: "+tn1.getTall());
        tn1.tellned();
        System.out.println("tn1: "+tn1.getTall());
        tn1.tellned();
        System.out.println("tn1: "+tn1.getTall());

        // Så kan vi lage en ny teller
        TellNed tn2 = new TellNed(100);
        System.out.println("tn2: "+tn2.getTall());

        tn1.tellned();
        tn2.tellned();
        System.out.println("tn1: "+tn1.getTall());
        System.out.println("tn2: "+tn2.getTall());

        tn1.tellned();
        tn2.tellned();
        System.out.println("tn1: "+tn1.getTall());
        System.out.println("tn2: "+tn2.getTall());

    }
}
